src/i18n/i18n.php

```diff
@@ -1970,256 +1956,167 @@ public static function get_cache()
+     * string is used and Entity identifies the string inside the namespace.
+     * @param mixed $arg,... Additional arguments are parsed as such:
+     *  - Next string argument is a default. Pass in a `|` pipe-delimited value with `{count}`
+     *    to do pluralisation.
+     *  - Any other string argument after default is context for i18nTextCollector
+     *  - Any array argument in any order is an injection parameter list. Pass in a `count`
+     *    injection parameter to pluralise.
+     * @return string
      */
-    protected static function with_translators($callback)
+    public static function _t($entity, $arg = null)
     {
-        // get current locale (either default or user preference)
-        $locale = i18n::get_locale();
-        $lang = i18n::get_lang_from_locale($locale);
-
-        // Only call getter if static isn't already defined (for performance reasons)
-        $translatorsByPrio = self::$translators ?: self::get_translators();
-
-        foreach ($translatorsByPrio as $priority => $translators) {
-            /** @var Zend_Translate $translator */
-            foreach ($translators as $name => $translator) {
-                $adapter = $translator->getAdapter();
-
-                // at this point, we need to ensure the language and locale are loaded
-                // as include_by_locale() doesn't load a fallback.
-
-                // TODO Remove reliance on global state, by refactoring into an i18nTranslatorManager
-                // which is instanciated by core with a $clean instance variable.
-
-                if (!$adapter->isAvailable($lang)) {
-                    i18n::include_by_locale($lang);
-                }
-
-                if (!$adapter->isAvailable($locale)) {
-                    i18n::include_by_locale($locale);
-                }
-
-                $result = call_user_func($callback, $adapter);
-                if ($result !== null) {
-                    return $result;
-                }
+        // Detect args
+        $default = null;
+        $injection = [];
+        foreach (array_slice(func_get_args(), 1) as $arg) {
+            if (is_array($arg)) {
+                $injection = $arg;
+            } elseif (!isset($default)) {
+                $default = $arg ?: '';
             }
         }
 
-        // Nothing matched
-        return null;
-    }
+        // Encourage the provision of default values so that text collector can discover new strings
+        if (!$default && static::config()->get('missing_default_warning')) {
+            user_error("Missing default for localisation key $entity", E_USER_WARNING);
+        }
 
+        // Deprecate legacy injection format (`string %s, %d`)
+        // inject the variables from injectionArray (if present)
+        $sprintfArgs = [];
+        if ($default && !preg_match('/\{[\w\d]*\}/i', $default) && preg_match('/%[s,d]/', $default)) {
+            Deprecation::notice('5.0', 'sprintf style localisation variables are deprecated');
```

Comment:
We also need to replace existing core uses: https://github.com/silverstripe/silverstripe-framework/issues/6582.